File: ws/Vaca/u1/WS08ChickenFarmSimulator/WS08ChickenFarmSimulator2/src/ec/edu/espe/chickenfarmsimulator/model/Chicken.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.chickenfarmsimulator.model;

import java.util.Date;
import java.text.SimpleDateFormat;

/**
 * Chicken Model
 * 
 * Author: Luis Vaca, TheJavaBandits, DCCO-ESPE
 */
public class Chicken {
    private int id;
    private String color;
    private String name;
    private int age;
    private boolean molting;
    private Date bornOnDate;
    
    public Chicken(int id, String name, String color, int age, boolean molting, Date bornOnDate) {
        this.id = id;
        this.color = color;
        this.name = name;
        this.age = age;
        this.molting = molting;
        this.bornOnDate = bornOnDate;
    }
    
    // Getters and Setters
    
    public int getId() {
        return id;
    }
    
    public void setId(int id) {
        this.id = id;
    }
    
    public String getColor() {
        return color;
    }
    
    public void setColor(String color) {
        this.color = color;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public int getAge() {
        return age;
    }
    
    public void setAge(int age) {
        this.age = age;
    }
    
    public boolean isMolting() {
        return molting;
    }
    
    public void setMolting(boolean molting) {
        this.molting = molting;
    }
    
    public Date getBornOnDate() {
        return bornOnDate;
    }
    
    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }
    
    @Override
    public String toString() {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
        return "Chicken{" + "id=" + id + ", name=" + name + ", color=" + color + ", age=" + age + ", molting=" + molting + ", bornOnDate=" + sdf.format(bornOnDate) + '}';
    }
}

File: hw/arboleda/u1/HW09CDToJavaScripAndPhyton/chicken_farm_simulator/ec_edu_espe_farmsimulator_model/Chicken.py
from datetime import datetime

class Chicken():
    #Constructor
    def __init__(self, id, name, color, age, molting, born_on_date):
   
        self.__id = id
        self.__name = name
        self.__color = color
        self.__age = age
        self.__molting = molting
        self.__born_on_date = born_on_date

    #toString()
    def __str__(self):
        return f"Chicken (id={self.__id}, name={self.__name}, color={self.__color}, age={self.__age}, molting={self.__molting}, born_on_date={self.__born_on_date})"

    #Getters
    def get_Id(self):
        return self.__id
    
    def get_Name(self):
        return self.__name
    
    def get_Color(self):
        return self.__color
    
    def get_Age(self):
        return self.__age
    
    def get_Molting(self):
        return self.__molting
    
    def get_Born_on_date(self):
        return self.__born_on_date
    
    #Setters
    def set_Id(self, id):
        self.__id=id
    
    def set_Name(self, name):
        self.__name=name
    
    def set_Color(self, color):
        self.__color=color
    
    def set_Age(self, age):
        self.__age=age
    
    def set_Molting(self, molting):
        self.__molting=molting
    
    def set_Born_on_date(self, born_on_date):
        self.__born_on_date=born_on_date
